Hippo.Web/ClientApp/src/components/Order/Details.tsx

```diff
@@ -0,0 +1,685 @@
+          setUpdateStatusModel({
+            currentStatus: order.status,
+            newStatus: "Submitted",
+          });
+          break;
+        case "Submitted":
+          setUpdateStatusModel({
+            currentStatus: order.status,
+            newStatus: "Processing",
+          });
+          break;
+        case "Processing":
+          setUpdateStatusModel({
+            currentStatus: order.status,
+            newStatus: "Active",
+          });
+          break;
+        default:
+          setUpdateStatusModel({
+            currentStatus: order.status,
+            newStatus: order.status,
+          });
+      }
+    }
+  }, [order]);
+
+  const historyColumnHelper = createColumnHelper<HistoryModel>();
+
+  const historyColumns = [
+    historyColumnHelper.accessor("actedDate", {
+      header: "Date",
+      id: "actedDate",
+      cell: (value) => (
+        <span>{convertToPacificTime(value.row.original.actedDate)}</span>
+      ),
+    }),
+    historyColumnHelper.accessor("actedBy", {
+      header: "Actor",
+      id: "actedBy",
+      cell: (value) => (
+        <>
+          {value.row.original.actedBy && (
+            <>
+              {value.row.original.actedBy.firstName}{" "}
+              {value.row.original.actedBy.lastName} (
+              {value.row.original.actedBy.email})
+            </>
+          )}
+          {!value.row.original.actedBy && <>System</>}
+        </>
+      ),
+    }),
+    historyColumnHelper.accessor("status", { header: "Status", id: "status" }),
+    historyColumnHelper.accessor("details", {
+      header: "Details",
+      id: "details",
+    }),
+  ];
+
+  const paymentColumnHelper = createColumnHelper<PaymentModel>();
+
+  const paymentColumns = [
+    paymentColumnHelper.accessor("amount", {
+      header: "Amount",
+      id: "amount",
+      cell: (value) => (
+        <span>
+          <i className="fas fa-dollar-sign" />{" "}
+          {value.row.original.amount.toFixed(2)}
+        </span>
+      ),
+    }),
+    paymentColumnHelper.accessor("status", { header: "Status", id: "status" }),
+    paymentColumnHelper.accessor("createdOn", {
+      header: "Created On",
+      id: "createdOn",
+      cell: (value) => (
+        <span>{convertToPacificTime(value.row.original.createdOn)}</span>
+      ),
+    }),
+    paymentColumnHelper.accessor("createdBy", {
+      header: "Created By",
+      id: "createdBy",
+      cell: (value) => (
+        <>
+          {value.row.original.createdBy && (
+            <>
+              {value.row.original.createdBy.firstName}{" "}
+              {value.row.original.createdBy.lastName} (
+              {value.row.original.createdBy.email})
+            </>
+          )}
+          {!value.row.original.createdBy && <>System</>}
+        </>
+      ),
+    }),
+  ];
+
+  // async function so the form can manage the loading state
+  const submitOrder = async (updatedOrder: OrderModel) => {
+    const editedOrder: OrderModel = {
```

Comment:
why do the copy field by field, what's the point of this code?

Reply:
#393